src/main/java/baritone/pathing/movement/movements/MovementDiagonal.java

```diff
@@ -125,17 +126,22 @@ public static void cost(CalculationContext context, int x, int y, int z, int des
             destWalkOn = destInto;
         } else {
             destWalkOn = context.get(destX, y - 1, destZ);
-            if (!MovementHelper.canWalkOn(context, destX, y - 1, destZ, destWalkOn)) {
+            boolean standingOnABlock = MovementHelper.mustBeSolidToWalkOn(context, x, y-1, z, context.get(x, y-1, z));
```

Comment:
let's save a BSI lookup call by caching the block at `(x, y - 1, z)`

https://github.com/cabaletta/baritone/blob/f76283ebfeb5bba7d53d76f727335de665131951/src/main/java/baritone/pathing/movement/movements/MovementDiagonal.java#L148

it's saved as `fromDown` a little bit later on, let's move that up to something like `IBlockState fromDownState = context.get(x, y - 1, z);`, which could go maybe right after `boolean frostWalker = false;`, then use that in both places?

Reply:
Maybe move it into the if-else part like `destWalkOn` so we don't look it up when taking the early return?

Reply:
Sure